docs/usage.md

```diff
@@ -0,0 +1,175 @@
+# Usage
+
+This is a very small set of narrative documentation about
+how to use the Hypothesis for Ruby API. It is in no way
+complete, but is here as a kind of place holder to give
+you enough information to get started and to make me
+write down enough to make sure that the thing I'm describing
+isn't actually a bad idea.
+
+## Core Concepts
+
+When writing a test using Hypothesis, you have three things:
+
+* The hypothesis you are testing
+* A set of "givens" - values that are required for you to
+  run the test.
+* A set of "assumptions" - things that need to be true in
+  order for the test to be valid.
+
+The givens come from *providers*, which describe a range of
```

Comment:
Vaguely wondering if "givers" or "sources" might be clearer as a name than "providers".

Reply:
All of the possible names for this are terrible. :-( [Some prior art on this discussion](https://github.com/HypothesisWorks/hypothesis-python/issues/804)

For the specific suggestions: "givers" sounds hella awkward, and the probelm with "sources" is that a) It's already used elsewhere in the API (though internally and b) They're not *really* a source of example, they're a description of how an example could be built and the source of the example is the test case itself.